src/inx/mod.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod error;
+use inx::{client::InxClient, Channel};
+use serde::{Deserialize, Serialize};
+
+pub use self::error::InxError;
+
+/// A builder to establish a connection to INX.
+#[must_use]
+#[derive(PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub struct InxConfig {
+    address: String,
+}
+
+impl InxConfig {
+    /// Creates a new [`InxConfig`]. The `address` is the address of the nodes INX interface.
+    pub fn new(address: String) -> Self {
+        Self { address }
+    }
+
+    /// Constructs an [`InxClient`] by consuming the [`InxConfig`].
+    pub async fn build(self) -> Result<InxClient<Channel>, InxError> {
```

Comment:
It's probably easier to understand if there's a static `build` function on `InxClient` instead (which consumes the config), so like this:
```rust
impl InxClient<Channel> {
    pub async fn build(inx_config: InxConfig) -> Result<Self, InxError> {...}
}
But that's something I can also change in another PR if you want.

Reply:
I would also not call it `build`, but `init`.